netket/operator/_fermion2nd/numba.py

```diff
@@ -155,6 +164,11 @@ def _flattened_kernel(  # pragma: no cover
         orb_idxs_list = np.split(orb_idxs, term_split_idxs)
         daggers_list = np.split(daggers, term_split_idxs)
 
+        
+        if constraint:
+            x_set = set() # set of all states in the hilbert space for quick lookup
+            for raw in x:
+                x_set.add(hash_raw(raw))
```

Comment:
this seems potentially very memory consuming ?

Reply:
Yes, I was not able to do it better. Without has it is extremely slow

Reply:
Guys, if we have a huge arrays of configurations, keeping an array of integers next to it will have a negligible added memory cost.

this is fine